idaes/core/solvers/petsc.py

```diff
@@ -500,8 +508,17 @@ def petsc_dae_by_time_element(
         between.construct()
 
     solve_log = idaeslog.getSolveLogger("petsc-dae")
-    regular_vars, time_vars = flatten_dae_components(m, time, pyo.Var)
-    regular_cons, time_cons = flatten_dae_components(m, time, pyo.Constraint)
+    if snes_options is not None:
+        logger = idaeslog.getIdaesLogger("idaes")
```

Comment:
We generally use `logger = idaeslog.getLogger(__name__)` for this - using `__name__` gives more granularity and avoids hard-coded string constants. Also, Pyomo has a deprecation warning function for things like this:

```
from pyomo.common.deprecation import deprecation_warning

deprecation_warning(
    msg=msg, logger=_log, version="2.0.0", remove_in="3.0.0"
)
```